### ISSUE TITLE
Wrong warnings about missing backticks in URLs

### ISSUE THREAD

**Issue by briansmith** [Manishearth/rust-clippy#1508]
In https://github.com/briansmith/ring/issues/285#issuecomment-277328045, @pietro wrote:
>  BTW, currently clippy warns about lack of backticks on documentation [...]. Some of the things clippy warns about are definitively wrong, like SpecialPublications in http://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Ar2.pdf

Example:

```rust
/// See [NIST SP 800-56A, revision 2].
///
/// [NIST SP 800-56A, revision 2]:
///     http://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Ar2.pdf
fn foo() { }
```

**Comment by Keats** [comment]
I believe it's the same as https://github.com/Manishearth/rust-clippy/issues/902

**Comment by briansmith** [comment]
Thanks! I'll move my comment over there. This should be marked as a duplicate.
